### PR TITLE
fix(template): update according to new discussed standards

### PR DESCRIPTION
**Pull request by neubs-bsi**
Issue Template changes:
- removed checkboxes from definition of done, since they added overhead

Pull Request Template changes:
- removed superfluous headers
- removed commit message part since they are part of the
  contributing.md not the pull request template
- added SW360 Integration Tests to possible tests list

CONTRIBUTING.md:
Add content to PR part about reviewing and squashing that
reflect the new standars.
Update style guide for commit messages to reflect the new standards

#### NOTE - Discuss
- I have considered removing the checklist from the pull request template, but found in my last two pull requests that it was actually helping me having a cleaner pull request. 
- I have however noticed, that the **Type Of Change** list to me seems superfluous. 

#### Request Reviewer
@sschuberth @bs-ondem @maxhbr @blaumeiser-at-bosch 

#### Type of Change
- [x] Other

#### Checklist
- [x] My code follows the style and guidelines of this project
- [x] I have self-reviewed my code 
- [x] I have updated the documentation accordingly to my changes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -146,34 +146,66 @@ The process described here has several goals:
...
+When changes are required fixup your commits or squash new commits based on requested changes.
+[Here you can find a guide on fixups and autosquashing](https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html).
+Do not forget to amend your commit messages along with the changes.
 
 ## Style Guides
 
 ### Git Commit Messages
 
-We use the commit style of the [SW360 project](https://github.com/eclipse/sw360/wiki/Dev-Semantic-Commits).
+We use the commit style of the [SW360 project](https://github.com/eclipse/sw360/wiki/Dev-Semantic-Commits) for commit message headlines.
```
Comment (maxhbr) [inline CONTRIBUTING.md]: Haven't we decided to not use this commit message style? I am very unsure.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -146,34 +146,66 @@ The process described here has several goals:
...
+If you have adhered to the style guide of the [commit messages](#git-commit-messages), you will have that content already.
+4. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing.
+Should a status check fail and you believe it is unrelated to your contribution, please write so in a comment in the PR.
+
+While the prerequisites above may be satisfied, the reviewer(s) may still ask you to do additional work or ask for changes before your PR is finally accepted.
+Since the reviewer knows the code very well, they might have ideas how to include features more elegantly or see inconsistencies compared to the code base.
+
+#### Reviews
+Once your pull request is being reviewed, there are some guidelines.
+1. Threads started during a review should only be marked as resolved by the started of the thread.
```
Comment (maxhbr) [inline CONTRIBUTING.md]:
```suggestion
1. Threads started during a review should only be marked as resolved by the starter of the thread.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -146,34 +146,66 @@ The process described here has several goals:
...
 
-We use the commit style of the [SW360 project](https://github.com/eclipse/sw360/wiki/Dev-Semantic-Commits).
+We use the commit style of the [SW360 project](https://github.com/eclipse/sw360/wiki/Dev-Semantic-Commits) for commit message headlines.
+When writing the commit message headline, keep in mind that the prefixes in the commit messages are meant to help when searching for changes.
+
+Additionally, you should add descriptions to your commit messages of
+- changes you have made to the code and
+- why those changes are necessary and implemented in that way specifically
+
+If there is an issue to your commit, link it with an "#-mention".
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply CONTRIBUTING.md]: or how about saying "... link it using keywords as described [here](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords)."

File: CONTRIBUTING.md (hunk 1 above)
Comment (sschuberth) [reply CONTRIBUTING.md]: I believe we had decided against it, yes, as it often takes up too much space in the commit message title. Personally, I like to refer to Git's own [submitting patches](https://github.com/git/git/blob/master/Documentation/SubmittingPatches#L43), which is also [what ORT refers to](https://github.com/heremaps/oss-review-toolkit/blob/master/CONTRIBUTING.md#git-history), to describe the style I believe we agreed on.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,36 +1,33 @@
...
-#### Commit Messages  
->    The commit messages should follow the syntax for [auto-closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords). 
->    It is preferred if you reference the issue in your commit message.
+> * Did you add or update any new dependencies that are required for your change?
 
-#### Request Reviewer
-> You can add desired reviewers here with an @mention.  
+### Request Reviewer
+> You can add desired reviewers here with an @mention.
 > For a full list of reviewers check out the *Committers* slot on [the project site](https://projects.eclipse.org/projects/technology.sw360.antenna/who).
```
Comment (maxhbr) [inline .github/pull_request_template.md]: How to find out, what the github-ids for the committers is? Maybe it would be good to have a list in the wiki for that.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -146,34 +146,89 @@ The process described here has several goals:
...
+3. Once you have integrated changes that were requested, you can dismiss the review of the reviewer and re-request a review.
+
+When integrating a reviewers comments into a pull request, please fix the issue in the respective commit where it appears,
+instead of in a new commit. 
+
+An approval of the PR means that the PR can be merged as is, without any changes still necessary.
+If the owner of a PR is an Eclipse committer of the project, the owner can merge their own PR once approved.
+Otherwise, the PR needs to be merged by an Eclipse committer of the project.
+
+#### Squashing
```
Comment (maxhbr) [inline CONTRIBUTING.md]: Do we want to say, that formatting changes should be in independent commits?

Comment (neubs-bsi) [reply CONTRIBUTING.md]: Hm. I don't think anyone will do formatting changes in a PR like this. 
Maybe I am wrong though...

Comment (maxhbr) [reply CONTRIBUTING.md]: I have seen formatting in many recent PRs (e.g. spaces after `,`). These should in my opinion be kept in a formatting commit

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -5,31 +5,22 @@
...
-- [ ] `mvn test` of whole project
-- [ ] `mvn test` of submodule
-- [ ] `gradle test` (for the gradle plugin)
-- [ ] Integration tests with a running SW360
+> If you have added any changes that require additional tests, or changes in tests, you should implement them and describe them here.  
+> All test that passed before your contribution should pass after it as well. 
 
 ### Checklist
 Must:
 - [ ] Existing tests still pass with my changes
```
Comment (sschuberth) [inline .github/pull_request_template.md]: I'd still argue that "[ ] Existing tests still pass with my changes" should also be removed because it's CI's responsibility to check for that.
